--- app/helpers/console/console_helper.rb	2013-07-17 09:39:25.239426127 +0000
+++ app/helpers/console/console_helper.rb	2013-07-17 09:40:12.258426117 +0000
@@ -9,10 +9,10 @@
   end
 
   def product_branding
-    content_tag(:span, "<strong>Open</strong>Shift Origin".html_safe, :class => 'brand-text headline')
+    content_tag(:span, "Zone<strong>52</strong>".html_safe, :class => 'brand-text headline')
   end
 
   def product_title
-    'OpenShift Origin'
+    'Zone52'
   end
 end
